File: push/src/main/java/com/xiaomi/push/service/timers/AlarmV21.java
package com.xiaomi.push.service.timers;

import android.app.job.JobInfo;
import android.app.job.JobScheduler;
import android.content.ComponentName;
import android.content.Context;
import android.os.SystemClock;
import com.xiaomi.channel.commonutils.logger.MyLog;
import com.xiaomi.push.service.XMJobService;
import com.xiaomi.push.service.timers.Alarm;
import com.xiaomi.smack.SmackConfiguration;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/service/timers/AlarmV21.class */
public class AlarmV21 implements Alarm.IAlarm {
    Context mContext;
    JobScheduler mJobScheduler;
    private boolean mStarted = false;

    AlarmV21(Context context) {
        this.mContext = context;
        this.mJobScheduler = (JobScheduler) context.getSystemService("jobscheduler");
    }

    @Override // com.xiaomi.push.service.timers.Alarm.IAlarm
    public boolean isAlive() {
        return this.mStarted;
    }

    void register(long j) {
        JobInfo.Builder builder = new JobInfo.Builder(1, new ComponentName(this.mContext.getPackageName(), XMJobService.class.getName()));
        builder.setMinimumLatency(j);
        builder.setOverrideDeadline(j);
        builder.setRequiredNetworkType(1);
        builder.setPersisted(false);
        MyLog.v("schedule Job = " + builder.build().getId() + " in " + j);
        this.mJobScheduler.schedule(builder.build());
    }

    @Override // com.xiaomi.push.service.timers.Alarm.IAlarm
    public void registerPing(boolean z) {
        if (z || this.mStarted) {
            long pingInteval = SmackConfiguration.getPingInteval();
            long jElapsedRealtime = pingInteval;
            if (z) {
                stop();
                jElapsedRealtime = pingInteval - (SystemClock.elapsedRealtime() % pingInteval);
            }
            this.mStarted = true;
            register(jElapsedRealtime);
        }
    }

    @Override // com.xiaomi.push.service.timers.Alarm.IAlarm
    public void stop() {
        this.mStarted = false;
        this.mJobScheduler.cancel(1);
    }
}

File: legacy/src/main/java/com/xiaomi/push/service/timers/AlarmV21.kt
package com.xiaomi.push.service.timers
import io.github.magisk317.mipush.protocol.model.*

import android.app.job.JobInfo
import android.app.job.JobScheduler
import android.content.ComponentName
import android.content.Context
import android.os.SystemClock
import com.xiaomi.channel.commonutils.logger.MyLog
import com.xiaomi.push.service.XMJobService
import com.xiaomi.smack.SmackConfiguration

class AlarmV21(private val mContext: Context) : Alarm.IAlarm {
    private val mJobScheduler: JobScheduler = mContext.getSystemService("jobscheduler") as JobScheduler
    private var mStarted = false

    override fun isAlive(): Boolean = mStarted

    private fun register(delayMillis: Long) {
        val builder = JobInfo.Builder(1, ComponentName(mContext.packageName, XMJobService::class.java.name))
        builder.setMinimumLatency(delayMillis)
        builder.setOverrideDeadline(delayMillis)
        builder.setRequiredNetworkType(1)
        builder.setPersisted(false)
        MyLog.v("schedule Job = ${builder.build().id} in $delayMillis")
        mJobScheduler.schedule(builder.build())
    }

    override fun registerPing(force: Boolean) {
        if (force || mStarted) {
            val pingInterval = SmackConfiguration.pingInterval.toLong()
            val elapsedRealtime = if (force) {
                stop()
                pingInterval - (SystemClock.elapsedRealtime() % pingInterval)
            } else {
                pingInterval
            }
            mStarted = true
            register(elapsedRealtime)
        }
    }

    override fun stop() {
        mStarted = false
        mJobScheduler.cancel(1)
    }
}
